packages/flutter/lib/src/material/text_field.dart

```diff
@@ -593,8 +598,16 @@ class TextField extends StatefulWidget {
   /// If [maxLength] is set, [maxLengthEnforced] indicates whether or not to
   /// enforce the limit, or merely provide a character counter and warning when
   /// [maxLength] is exceeded.
+  @Deprecated(
+    'Use maxLengthEnforcement parameter which provide more specific '
+    'behaviour related max length limitation. '
+    'This feature was deprecated after v1.23.0-19.0.pre.'
+  )
   final bool maxLengthEnforced;
 
+  /// {@macro flutter.services.textFormatter.maxLengthEnforcement}
```

Comment:
The developer won't be able to confidently determine which `MaxLengthEnforcement` option should be used by just looking at this section alone (unless they know ios and android very very well), and they might go for the `default` option thinking it's more CJK friendly, and find out later that it doesn't work well with Gboard. I think we should document how Gboard and ios IME use composing text differently.

Reply:
Added more description.